Command/CacheCommandTrait.php

```diff
@@ -0,0 +1,142 @@
+ */
+
+namespace Liip\ImagineBundle\Command;
+
+use Imagine\Exception\RuntimeException;
+use Liip\ImagineBundle\Component\Console\Style\ImagineStyle;
+use Liip\ImagineBundle\Imagine\Cache\CacheManager;
+use Liip\ImagineBundle\Imagine\Filter\FilterManager;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * @internal
+ */
+trait CacheCommandTrait
+{
+    /**
+     * @var CacheManager
+     */
+    private $cacheManager;
+
+    /**
+     * @var FilterManager
+     */
+    private $filterManager;
+
+    /**
+     * @var ImagineStyle
+     */
+    private $io;
+
+    /**
+     * @var bool
+     */
+    private $outputMachineReadable;
+
+    /**
+     * @var int
+     */
+    private $failures = 0;
+
+    private function setupOutputStyle(InputInterface $input, OutputInterface $output): void
+    {
+        $this->outputMachineReadable = $input->getOption('as-script');
+        $this->io = new ImagineStyle($input, $output, $this->outputMachineReadable ? false : !$input->getOption('no-colors'));
+    }
+
+    /**
+     * @return array[]
+     */
+    private function resolveInputFiltersAndPaths(InputInterface $input): array
+    {
+        return [
+            $input->getArgument('path'),
+            $this->normalizeFilterList($input->getOption('filter')),
+        ];
+    }
+
+    /**
+     * @param string[] $filters
+     *
+     * @return string[]
+     */
+    private function normalizeFilterList(array $filters): array
+    {
+        if (0 < count($filters)) {
+            return $filters;
+        }
+
+        if (0 < count($filters = array_keys((array) $this->filterManager->getFilterConfiguration()->all()))) {
+            return $filters;
+        }
+
+        throw new RuntimeException('No filters have been defined in the active configuration!');
+    }
+
+    private function outputCommandHeader(): void
+    {
+        if (!$this->outputMachineReadable) {
+            $this->io->title($this->getName(), 'liip/imagine-bundle');
+        }
+    }
+
+    /**
+     * @param string[] $images
+     * @param string[] $filters
+     */
+    private function outputCommandResult(array $images, array $filters): void
+    {
+        if (!$this->outputMachineReadable) {
+            $wordPluralizer = function (int $count, string $singular) {
+                return 1 === $count ? $singular : sprintf('%ss', $singular);
+            };
+            $rootTextFormat = 'Completed %d %s (%d %s, %d %s)';
+            $imagePathsSize = count($images);
+            $filterSetsSize = count($filters);
+            $allActionsSize = ($filterSetsSize * $imagePathsSize) - $this->failures;
+            $allActionsWord = $wordPluralizer($allActionsSize, 'operation');
+
+            switch (substr(__CLASS__, -18, 6)) {
+                case 'Remove':
+                    $allActionsWord = $wordPluralizer($allActionsSize, 'removal');
+                    break;
+
+                case 'esolve':
```

Comment:
esolve? should this be Resolve?

Reply:
It was a little bit weird, I changed the method a little bit

Reply:
thanks